test/org/zaproxy/zap/extension/pscanrulesAlpha/XBackendServerInformationLeakUnitTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2018 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.junit.Before;
+import org.junit.Test;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+public class XBackendServerInformationLeakUnitTest extends PassiveScannerTest {
+
+	private static final String XBS_HEADER = "X-Backend-Server";
+	private static final String HEADER_VALUE = "developer1.webapp.scl3.mozilla.com";
+	private HttpMessage msg;
+
+	@Before
+	public void before() throws URIException {
+		HttpRequestHeader requestHeader = new HttpRequestHeader();
+		requestHeader.setURI(new URI("http://example.com", false));
+
+		msg = new HttpMessage();
+		msg.setRequestHeader(requestHeader);
+	}
+
+	@Override
+	protected PluginPassiveScanner createScanner() {
+		return new XBackendServerInformationLeak();
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfResponseHasNoRelevantContent() {
+		// Given
+		msg.setResponseBody("Some text <h1>Some Title Element</h1>");
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void shouldRaiseAlertIfResponseHasRelevantContent() {
+		// Given
+		msg.getResponseHeader().addHeader(XBS_HEADER, HEADER_VALUE);
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// THen
+		assertThat(alertsRaised.size(), equalTo(1));
+		assertThat(alertsRaised.get(0).getEvidence(), equalTo(HEADER_VALUE));
+	}
+
+	@Test
+	public void shouldRaiseAlertIfResponseHasRelevantContentIgnoringCase() {
```

Comment:
This check is not consistent, just `XBackendServerInformationLeakUnitTest` and `ServerHeaderInfoLeakScannerUnitTest` are testing with different case. Is there are specific reason to add to these two but not the others?

Reply:
Not really, it probably just depends what a modeled things based on.

I think this test variant can be removed, the header retrieval forces upper or lower (I forget which) during comparison....

Reply:
Sounds good.